injected/src/features/duck-ai-data-clearing.js

```diff
@@ -0,0 +1,92 @@
+import ContentFeature from '../content-feature.js';
+
+/**
+ * This feature is responsible for clearing Duck.ai-related data when the `duckAiClearData`
+ * message is received. It clears the `savedAIChats` item from localStorage and the `chat-images`
+ * object store from IndexedDB, then sends a `duckAiClearDataCompleted` message if successful
+ * or a `duckAiClearDataFailed` message if either call is unsuccessful.
+ */
+export class DuckAiDataClearing extends ContentFeature {
+    init() {
+        this.messaging.subscribe('duckAiClearData', (_) => this.clearData());
+    }
+
+    async clearData() {
+        let success = true;
+
+        try {
+            this.clearSavedAIChats();
+        } catch (error) {
+            success = false;
+            this.log.error('Error clearing `savedAIChats`:', error);
+        }
+
+        try {
+            await this.clearChatImagesStore();
+        } catch (error) {
+            success = false;
+            this.log.error('Error clearing `chat-images` object store:', error);
+        }
+
+        if (success) {
+            this.notify('duckAiClearDataCompleted');
+        } else {
+            this.notify('duckAiClearDataFailed');
+        }
+    }
+
+    clearSavedAIChats() {
+        this.log.info('Clearing `savedAIChats`');
+
+        window.localStorage.removeItem('savedAIChats');
+    }
+
+    clearChatImagesStore() {
+        this.log.info('Clearing `chat-images` object store');
+
+        return new Promise((resolve, reject) => {
+            const request = window.indexedDB.open('savedAIChatData');
```

Comment:
Do these keys need to be configurable like the Android ones are?

You can just use `getFeatureSetting` api to get the value from remote config.

I think it'd be good to do this for any constants in this function.